homeassistant/components/sensor/waqi.py

```diff
@@ -0,0 +1,85 @@
+"""
+Support for the World Air Quality Index service.
+
+For more details about this platform, please refer to the documentation at
+https://github.com/home-assistant/home-assistant/
+"""
+import logging
+from datetime import timedelta
+from homeassistant.helpers.entity import Entity
+
+
+REQUIREMENTS = ["pwaqi==1.2"]
+
+_LOGGER = logging.getLogger(__name__)
+
+SENSOR_TYPES = {
+    'aqi':          ['AQI', '0-300+', 'mdi:cloud']
+}
+
+ATTR_LOCATION = 'locations'
+
+# Return cached results if last scan was less then this time ago
+# Cyprus Air Quality data is uploaded to server every 10mn
+# so this time should not be under
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=600)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the requested World Air Quality Index locations."""
+    dev = []
+    try:
+        import pwaqi
+        _LOGGER.info('Known locations: %s', config[ATTR_LOCATION])
+
+        # Iterate each module
+        for location_name in config[ATTR_LOCATION]:
+            _LOGGER.info('Adding location %s', location_name)
+            station_ids = pwaqi.findStationCodesByCity(location_name)
+            _LOGGER.info('I got the following stations: %s', station_ids)
+            for station in station_ids:
+                dev.append(WaqiSensor(station))
+    except KeyError:
```

Comment:
Instead of expecting a key error, include a `PLATFORM_SCHEMA` that will enforce that the key is present.